Code/Framework/AzCore/AzCore/IO/Streamer/Statistics.h

```diff
@@ -10,127 +10,297 @@
-                m_values[index] = value;
-                ++m_count;
-            }
+        static Statistic CreateByteSize(
+            AZStd::string_view owner,
+            AZStd::string_view name,
+            u64 value,
+            AZStd::string_view description = "",
+            GraphType graphType = GraphType::Histogram);
+
+        static Statistic CreateByteSizeRange(
+            AZStd::string_view owner,
+            AZStd::string_view name,
+            u64 value,
+            u64 min,
+            u64 max,
+            AZStd::string_view description = "",
+            GraphType graphType = GraphType::Histogram);
+
+        static Statistic CreateTime(
+            AZStd::string_view owner,
+            AZStd::string_view name,
+            TimeValue value,
+            AZStd::string_view description = "",
+            GraphType graphType = GraphType::Lines);
+
+        static Statistic CreateTimeRange(
+            AZStd::string_view owner,
+            AZStd::string_view name,
+            TimeValue value,
+            TimeValue min,
+            TimeValue max,
+            AZStd::string_view description = "",
+            GraphType graphType = GraphType::Lines);
+
+        static Statistic CreateBytesPerSecond(
+            AZStd::string_view owner,
+            AZStd::string_view name,
+            double value,
+            AZStd::string_view description = "",
+            GraphType graphType = GraphType::Histogram);
+
+        static Statistic CreatePermanentString(
+            AZStd::string_view owner, AZStd::string_view name, AZStd::string value, AZStd::string_view description = "");
+        static Statistic CreateReferenceString(
+            AZStd::string_view owner, AZStd::string_view name, AZStd::string_view value, AZStd::string_view description = "");
+        static void PlotImmediate(AZStd::string_view owner, AZStd::string_view name, double value);
+
+        AZStd::string_view GetOwner() const;
+        AZStd::string_view GetName() const;
+        AZStd::string_view GetDescription() const;
+        const Value& GetValue() const;
+        GraphType GetGraphType() const;
 
-            //! Calculates the average value for the window.
-            AverageType CalculateAverage() const
+    private:
+        Statistic() = default;
+
+        template<typename T>
+        static Statistic Create(
+            AZStd::string_view owner, AZStd::string_view name, T&& value, AZStd::string_view description, GraphType graphType);
+
+        Value m_value;
+        AZStd::string_view m_owner;
+        AZStd::string_view m_name;
+        AZStd::string_view m_description;
+        GraphType m_graphType;
+    };
+
+    //! AverageWindow keeps track of the average of values in a sliding window.
+    //! @StorageType The type of the value in the sliding window. Need to be a number of a AZStd::chrono::duration.
+    //! @AverageType The type CalculateAverage will return. StorageType needs to be able to be converted to AverageType.
```

Comment:
Can this pre-condition about conversion of `StorageType` to `AverageType` be expressed with additional type trait `static_assert`s below?

Reply:
That's indeed possible and a good idea. Done in the next commit.